### ISSUE TITLE
all test passed, but when run traincurfil_train it will be an runtime_error.

### ISSUE THREAD

**Issue by tornadomeet** [deeplearningais/curfil#13]
hello @temporaer ,
i installed curfil correctly, and all test has passed when i use `ctest`command, here is the log:

```
    Start 1: image_test
1/7 Test #1: image_test .......................   Passed    1.11 sec
    Start 2: feature_generation_test
2/7 Test #2: feature_generation_test ..........   Passed    1.98 sec
    Start 3: random_tree_test
3/7 Test #3: random_tree_test .................   Passed    0.02 sec
    Start 4: random_tree_image_gpu_test
4/7 Test #4: random_tree_image_gpu_test .......   Passed    8.56 sec
    Start 5: image_cache_test
5/7 Test #5: image_cache_test .................   Passed    0.48 sec
    Start 6: import_export_test
6/7 Test #6: import_export_test ...............   Passed    1.47 sec
    Start 7: random_tree_image_test
7/7 Test #7: random_tree_image_test ...........   Passed   37.40 sec

100% tests passed, 0 tests failed out of 7
```

but when i use `curfil_train` to train the nyu_depth_v2 dataset(has been converted to curfil format), it outputs one run time error,like this:

```
terminate called after throwing an instance of 'std::runtime_error'
  what():  cudaMalloc3DArray(&colorTextureData, &channelDesc, extent, flags): invalid argument
```

i just debug the code, and find it will break(just the same error infor) in all 3D-aware function with cuda, like cudaMalloc3DArray,cudaMemcpy3DAsync..., so how to fix this?thanks.
my gpu is K80, cuda 7.0, and i used device 0 for training. i also set `SET( CUDA_ARCHITECTURE -gencode arch=compute_37,code=sm_37 )` in src/CMakeLists.txt.

**Comment by temporaer** [comment]
when you debug the function, do the parameters to `cudaMalloc3DArray` make sense? If yes, I can think of three options: gpu isn't initialized properly, too large 3D caches allocated, or a change in cuda interface (since I never ran this lib with cuda 7.0). My guess would be the size, maybe you can compare your card specs with the value passed to the alloc function?

**Comment by tornadomeet** [comment]
@temporaer  thanks a lot! i have solved this by your info: just **decrease the value of -- imageCacheSize**.  

   But i don't get the real reason why should we do that, i have done some tests and i fond if we don't set imageCacheSize (inside the code, it will auto make 0.66 of the free memory on GPU)when using curfil_train, then it will failed like above; and if i set --imageCacheSize 5120 on my k80(12G memory), it will failed too,but  if we set it less than 4096, it will work fine.

i debug the code:
```
{
        cudaChannelFormatDesc channelDesc = cudaCreateChannelDesc(32, 0, 0, 0, cudaChannelFormatKindFloat);
        cudaExtent extent = make_cudaExtent(width, height, colorChannels * getCacheSize());
        cudaSafeCall(cudaMalloc3DArray(&colorTextureData, &channelDesc, extent, flags));
    }

    {
        cudaChannelFormatDesc channelDesc = cudaCreateChannelDesc(32, 0, 0, 0, cudaChannelFormatKindSigned);
        cudaExtent extent = make_cudaExtent(width, height, depthChannels * getCacheSize());
        cudaSafeCall(cudaMalloc3DArray(&depthTextureData, &channelDesc, extent, flags));
    }
```

if we set imageCacheSize too large(such as 5120), it means that all nyu_depth_v2 images(795) will be put in image cache, so getCacheSize()=795 and 
colorTextureData size = 640*480*3*795*4/1024/1024=2795MB;
depthTextureData size = 640*480*3*795/1024/1024=932MB;
all of them is only 3727MB;
**this is less than the imageCacheSize which we set and my GPU also can provide that size memory**,  so why it will crashed for cudaMalloc3DArray? or how to make the code give erro/warning message ahead of running the code(such as using assert() in code)? thanks!
